### PR TITLE
SORQA-355

### PR DESCRIPTION
**Pull request by lb-sgent**
image

### CODE REVIEW

**Review by pk-sgent — APPROVED** [review]

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java [hunk 1]
```diff
@@ -722,6 +679,59 @@ public CreateNewSampleSteps(
...
+        "I check is Sent dispatched Date and Received Date fields required",
+        () -> {
+          webDriverHelpers.waitUntilIdentifiedElementIsPresent(DATE_SAMPLE_SEND);
+
+          webDriverHelpers.waitUntilIdentifiedElementIsPresent(DATE_SAMPLE_RECEIVED);
+        });
+
+    And(
+        "I click Add Pathogen test in Sample creation page in German",
+        () -> webDriverHelpers.clickOnWebElementBySelector(ADD_PATHOGEN_TEST_BUTTON_DE));
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java:706]: I know is out of scope for this PR, but can we make the add pathogen test button generic? to work for both international and german markets?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java [hunk 2]
```diff
@@ -722,6 +679,59 @@ public CreateNewSampleSteps(
...
+
+    And(
+        "I check DATE AND TIME OF RESULT field",
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java:709]: this is not a valid check.
1. When you check something, you should use assert and know exactly what to check.
2. Add also place of where action is done
3. Why sleep?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java [hunk 3]
```diff
@@ -722,6 +679,59 @@ public CreateNewSampleSteps(
...
+        () -> {
+          TimeUnit.SECONDS.sleep(10);
+          webDriverHelpers.getWebElement(DATE_AND_TIME_OF_RESULTS_DE);
+        });
+
+    And(
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java:715]: please check if this method doesn't already exist

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java [hunk 4]
```diff
@@ -722,6 +679,59 @@ public CreateNewSampleSteps(
...
+        "I click on save sample button",
+        () -> {
+          webDriverHelpers.clickOnWebElementBySelector(SAVE_SAMPLE_BUTTON);
+        });
+
+    Then(
+        "I check error popup message in German",
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java:722]: as commented on created method, just get the list of elements, and use assert for this

**Review by rdutu-vg — CHANGES_REQUESTED** [review]

File: sormas-e2e-tests/src/test/resources/features/sanity/web/Sample.feature [hunk 5]
```diff
@@ -261,4 +261,24 @@ Feature: Sample Functionalities
     Then I set Tested disease variant as "B.1.617.3"
     And I save the created sample
     And I confirm update case result
-    Then I check if Update case disease variant popup is available
\ No newline at end of file
+    Then I check if Update case disease variant popup is available
+
+  @issue=SORDEV-7427 @env_de
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/resources/features/sanity/web/Sample.feature:266]: replace issue with tmsLink

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java (hunk 3 above)
Comment (lb-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java:715]: All methods with SAVE_SAMPLE_BUTTON in CreateNewSampleSteps contains some other actions. There is no simple "click on save button" method.

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java (hunk 4 above)
Comment (lb-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java:722]: Added assert in checkIsPopupContainsList method.

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/helpers/WebDriverHelpers.java [hunk 6]
```diff
@@ -1022,4 +1022,15 @@ public void closeActiveWindow() {
           "Cannot close active window and switch to parent window because only one is available!");
     }
   }
+
+  public void checkIsPopupContainsList(By popup, List popupElements) {
```
[the remark this answers is not in the document]
Comment (lb-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/helpers/WebDriverHelpers.java:1026]: There is no other method that can check message from error popup and compare it with expected message.
Also added assert to method.

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java [hunk 7]
```diff
@@ -263,48 +268,6 @@ public CreateNewSampleSteps(
           webDriverHelpers.clickOnWebElementBySelector(SAVE_SAMPLE_BUTTON);
         });
 
-    When(
```
[the remark this answers is not in the document]
Comment (lb-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java:266]: Previous I was writing test with wrong scenario understanding. Changed after call with Pawel.

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java [hunk 8]
```diff
@@ -263,48 +268,6 @@ public CreateNewSampleSteps(
...
-        (String diseaseType, String testType) -> {
-          sample = sampleService.buildPathogenTestResultTypeVerified(diseaseType, testType);
-          selectTypeOfTest(sample.getTypeOfTest());
-          selectTestedDisease(sample.getTestedDisease());
-          selectTestResult(sample.getSampleTestResults());
-          selectResultVerifiedByLabSupervisor(
-              sample.getResultVerifiedByLabSupervisor(), RESULT_VERIFIED_BY_LAB_SUPERVISOR_OPTIONS);
-        });
-
-    When(
```
[the remark this answers is not in the document]
Comment (lb-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java:277]: Previous I was writing test with wrong scenario understanding. Changed after call with Pawel

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java [hunk 9]
```diff
@@ -263,48 +268,6 @@ public CreateNewSampleSteps(
...
-          selectTestedDisease(sample.getTestedDisease());
-          selectTypeOfTest(sample.getTypeOfTest());
-          selectTestResult(sample.getSampleTestResults());
-          fillDateOfResult(sample.getDateOfResult(), Locale.ENGLISH);
-          selectLaboratory(sample.getLaboratory());
-          selectResultVerifiedByLabSupervisor(
-              sample.getResultVerifiedByLabSupervisor(), RESULT_VERIFIED_BY_LAB_SUPERVISOR_OPTIONS);
-        });
-
-    When(
```
[the remark this answers is not in the document]
Comment (lb-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java:296]: Previous I was writing test with wrong scenario understanding. Changed after call with Pawel

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java [hunk 10]
```diff
@@ -263,48 +268,6 @@ public CreateNewSampleSteps(
...
-        "I set PCR RT PCR Test specification to {string} option",
-        (String option) -> {
-          webDriverHelpers.selectFromCombobox(PCR_TEST_SPECIFICATION_COMBOBOX_DIV, option);
-        });
-
-    When(
```
[the remark this answers is not in the document]
Comment (lb-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java:302]: Previous I was writing test with wrong scenario understanding. Changed after call with Pawel

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java [hunk 11]
```diff
@@ -679,12 +642,6 @@ public CreateNewSampleSteps(
           webDriverHelpers.clickOnWebElementBySelector(CONFIRM_BUTTON);
         });
 
-    When(
```
[the remark this answers is not in the document]
Comment (lb-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java:682]: Previous I was writing test with wrong scenario understanding. Changed after call with Pawel

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java [hunk 12]
```diff
@@ -722,6 +679,59 @@ public CreateNewSampleSteps(
...
+
+    And(
+        "I select Received checkbox",
+        () -> {
+          webDriverHelpers.waitUntilElementIsVisibleAndClickable(SAMPLE_RECEIVED_CHECKBOX);
+          webDriverHelpers.clickOnWebElementBySelector(SAMPLE_RECEIVED_CHECKBOX);
+        });
+
+    Then(
+        "I check is Sent dispatched Date and Received Date fields required",
```
[the remark this answers is not in the document]
Comment (lb-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java:697]: I am checking attribute responsible for that is field mandatory or not

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java (hunk 1 above)
Comment (lb-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java:706]: It is reusing previous created variables and it need to refactor other methods using this variable.

**Pull request merged**